pkg/pushsync/pushsync.go

```diff
@@ -365,13 +397,23 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, retryAllo
 			if r.err != nil && r.attempted {
 				ps.metrics.TotalFailedSendAttempts.Inc()
 
-				// if the node has warmed up AND no other closer peer has been tried
-				if time.Now().After(ps.warmupPeriod) && !ps.skipList.HasChunk(ch.Address()) && ps.topologyDriver.IsWithinDepth(ch.Address()) {
-					ps.skipList.Add(peer, ch.Address(), skipPeerExpiration)
+				// if we have received 1 more results than peers attempted, return
+				// this means all attempts including establishing storage have already failed
+				if results > attempts {
+					return nil, ErrNoPush
 				}
 			}
 		case <-ctx.Done():
 			return nil, ctx.Err()
+		case <-sendReceipt:
+			// to make the loop wait for the one extra result because of this one time interruption, increase attempt exhaustion limits
+			allowedRetries++
+			i++
+			// if not in progress yet, establish storage
+			if !signInProgress {
+				signInProgress = true
+				go ps.establishStorage(ch, origin, resultC)
+			}
```

Comment:
or `goto` to top of the `select` which gets called after `go ps.establishStorage(ch, origin, resultC)`

Reply:
we achieve the same result by skipping the push peer block and entering the select.
intention was to avoid goto to make reasoning about the sequence of events happening in the loop more accessible